diff --git a/Dockerfiles/build-base/alpine/Dockerfile b/Dockerfiles/build-base/alpine/Dockerfile
index b3e54da05..1145a08e9 100644
--- a/Dockerfiles/build-base/alpine/Dockerfile
+++ b/Dockerfiles/build-base/alpine/Dockerfile
@@ -27,6 +27,7 @@ RUN set -eux && \
     ARCH_SUFFIX="$(apk --print-arch)"; \
     case "$ARCH_SUFFIX" in \
         ppc64le) openjdk_packages='openjdk11 openjdk11-jre-headless' ;; \
+        armv7)   openjdk_packages='' ;; \
         *)       openjdk_packages='openjdk17 openjdk17-jre-headless' ;; \
     esac; \
     INSTALL_PKGS="bash \
diff --git a/Dockerfiles/build-mysql/alpine/Dockerfile b/Dockerfiles/build-mysql/alpine/Dockerfile
index 9bf307543..145d7172c 100644
--- a/Dockerfiles/build-mysql/alpine/Dockerfile
+++ b/Dockerfiles/build-mysql/alpine/Dockerfile
@@ -91,6 +91,11 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
     set -eux && \
     cd "${ZBX_SOURCES_DIR}" && \
     ./bootstrap.sh && \
+    ARCH_SUFFIX="$(apk --print-arch)"; \
+    case "$ARCH_SUFFIX" in \
+        armv7) java_component='' ;; \
+        *)     java_component='--enable-java' ;; \
+    esac; \
     ./configure \
         --datadir=/usr/lib \
         --libdir=/usr/lib/zabbix \
@@ -99,7 +104,6 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
         --enable-ipv6 \
         --enable-agent \
         --enable-agent2 \
-        --enable-java \
         --enable-proxy \
         --enable-server \
         --enable-webservice \
@@ -115,6 +119,7 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
         --with-openssl \
         --with-ssh \
         --with-unixodbc \
+        $java_component \
         --silent && \
     jobs="${BUILD_JOBS:-$(nproc)}" && \
     make -j"${jobs}" -s dbschema && \
diff --git a/Dockerfiles/build-sqlite3/alpine/Dockerfile b/Dockerfiles/build-sqlite3/alpine/Dockerfile
index 86fe2666b..81b69af96 100644
--- a/Dockerfiles/build-sqlite3/alpine/Dockerfile
+++ b/Dockerfiles/build-sqlite3/alpine/Dockerfile
@@ -86,6 +86,11 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
     set -eux && \
     cd "${ZBX_SOURCES_DIR}" && \
     ./bootstrap.sh && \
+    ARCH_SUFFIX="$(apk --print-arch)"; \
+    case "$ARCH_SUFFIX" in \
+        armv7) java_component='' ;; \
+        *)     java_component='--enable-java' ;; \
+    esac; \
     ./configure \
         --datadir=/usr/lib \
         --libdir=/usr/lib/zabbix \
@@ -94,7 +99,6 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
         --enable-ipv6 \
         --enable-agent \
         --enable-agent2 \
-        --enable-java \
         --enable-proxy \
         --with-ares \
         --with-ldap \
@@ -108,6 +112,7 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
         --with-openssl \
         --with-ssh \
         --with-unixodbc \
+        $java_component \
         --silent && \
     jobs="${BUILD_JOBS:-$(nproc)}" && \
     make -j"${jobs}" -s dbschema && \
